BDArmory/Weapons/BDExplosivePart.cs

```diff
@@ -8,20 +8,28 @@
 using BDArmory.Settings;
 using BDArmory.Utils;
 using BDArmory.Weapons.Missiles;
+using BDArmory.Competition;
 
 namespace BDArmory.Weapons
 {
     public class BDExplosivePart : PartModule
     {
         float distanceFromStart = 500;
 
+        public BDTeam sourceteam
+        {
+            get { return _sourceteam; }
+            set { _sourceteam = value; SourceVesselTeam = _sourceteam != null ? _sourceteam.Name : null; }
+        }
+        BDTeam _sourceteam;
+
         public Vessel sourcevessel
         {
-            get { return _sourceVessel; }
-            set { _sourceVessel = value; SourceVesselName = _sourceVessel != null ? _sourceVessel.vesselName : null; }
+            get { return _sourcevessel; }
+            set { _sourcevessel = value; }
         }
-        Vessel _sourceVessel;
-        public string SourceVesselName { get; private set; }
```

Comment:
Is there a reason for removing `SourceVesselName`? The point of it was to preserve the vessel name in case the vessel is `null` by the time the explosions happen, which would give NREs trying to access `sourcevessel.GetName()`.

Reply:
I think it was a case of thinking it wasn't necessary if ExplosionFX was only going to be looking at sourceVesselTeam. Easy enough to revert and re-add